spec/services/participant_validation_service_spec.rb

```diff
@@ -249,4 +249,296 @@ def build_validation_result(trn:, options: {})
       previous_induction: false,
     }.merge(options)
   end
+
+  describe "#validate with actual DQT api", with_feature_flags: { actual_dqt_api: "active" } do
+    let(:trn) { "1234567" }
+    let(:nino) { "QQ123456A" }
+    let(:full_name) { "John Smith" }
+    let(:dob) { Date.new(1970, 1, 2) }
+    let(:qts_date) { 6.weeks.ago.to_date }
+    let(:alert) { false }
+    let(:dqt_record) do
+      {
+        trn: trn,
+        ni_number: nino,
+        name: full_name,
+        dob: dob,
+        active_alert: alert,
+        qualified_teacher_status: {
+          "qts_date" => qts_date,
+        },
+      }.stringify_keys
+    end
+    let(:validation_result) { ParticipantValidationService.validate(trn: trn, nino: nino, full_name: full_name, date_of_birth: dob) }
+
+    let(:mock_client) { instance_double(ActualDqt::Client, get_record: nil) }
+
+    before do
+      allow(DqtApiAccess).to receive(:token).and_return("jwt-access-token")
+      allow(ActualDqt::Client).to receive(:new).and_return(mock_client)
+    end
+
+    it "calls get_record on the DQT API client" do
+      expect(mock_client).to receive(:get_record).with(
+        trn: trn,
+        birthdate: dob,
+        nino: nino,
+      )
+
+      ParticipantValidationService.validate(trn: trn, nino: nino, full_name: full_name, date_of_birth: dob)
+    end
+
+    context "when the participant cannot be found" do
+      before do
+        expect(mock_client).to receive(:get_record).and_return(nil)
+      end
+
+      it "returns nil" do
+        expect(validation_result).to eql nil
+      end
+    end
+
+    context "when trn is less than 7 characters" do
+      let(:trn) { "123456" }
+      let(:padded_trn) { "0123456" }
+
+      let(:dqt_record) do
+        {
+          trn: "0123456", # API sends padded TRNs
+          ni_number: nino,
+          name: full_name,
+          dob: dob,
+          active_alert: alert,
+          qualified_teacher_status: {
+            "qts_date" => qts_date,
+          },
+        }.stringify_keys
+      end
+
+      let(:validation_result) do
+        ParticipantValidationService.validate(
+          trn: trn,
+          nino: "WRONG",
+          full_name: full_name,
+          date_of_birth: dob,
+        )
+      end
+
+      before do
+        expect(mock_client).to receive(:get_record).and_return(dqt_record)
+      end
+
+      it "returns record with padded trn when trn is not padded" do
+        expect(validation_result).to eql({
+          trn: padded_trn,
+          qts: true,
+          active_alert: false,
+          previous_induction: true,
```

Comment:
I'm not sure what these extra fields (`previous_induction`, `previous_participation`) are and appeared after i rebased. I just copied the tests across. it would be great if someone can confirm if these values are what they should be.